pkg/doc/vc/status/csl/csl.go

```diff
@@ -112,57 +119,75 @@ func (c *CredentialStatusManager) CreateStatusID() (*verifiable.TypedID, error)
 		}
 	}
 
-	return &verifiable.TypedID{ID: cslWrapper.CSL.ID, Type: CredentialStatusType}, nil
+	return &verifiable.TypedID{ID: cslWrapper.VC.ID + "#" + revocationListIndex,
+		Type: revocationList2020Status, CustomFields: verifiable.CustomFields{"revocationListIndex": revocationListIndex,
+			"revocationListCredential": cslWrapper.VC.ID}}, nil
 }
 
 // UpdateVCStatus update vc status
-func (c *CredentialStatusManager) UpdateVCStatus(v *verifiable.Credential, profile *vcprofile.DataProfile,
-	status, statusReason string) error {
-	cslWrapper, err := c.getCSLWrapper(v.Status.ID)
+func (c *CredentialStatusManager) RevokeVC(v *verifiable.Credential, profile *vcprofile.DataProfile) error {
+	revocationListCredential, ok := v.Status.CustomFields["revocationListCredential"].(string)
+	if !ok {
+		return fmt.Errorf("failed to cast vc subject revocationListCredential")
+	}
+
+	cslWrapper, err := c.getCSLWrapper(revocationListCredential)
 	if err != nil {
 		return err
 	}
 
-	signOpts, err := prepareSigningOpts(profile, v.Proofs)
+	signOpts, err := prepareSigningOpts(profile, cslWrapper.VC.Proofs)
 	if err != nil {
 		return err
 	}
 
-	statusCredential, err := c.createStatusCredential(v, status, statusReason)
+	cs, ok := cslWrapper.VC.Subject.([]verifiable.Subject)
+	if !ok {
+		return fmt.Errorf("failed to cast vc subject")
+	}
+
+	bitString, err := utils.DecodeBits(cs[0].CustomFields["encodedList"].(string))
 	if err != nil {
 		return err
 	}
 
-	signedStatusCredential, err := c.crypto.SignCredential(profile, statusCredential, signOpts...)
+	revocationListIndex, err := strconv.Atoi(v.Status.CustomFields["revocationListIndex"].(string))
```

Comment:
@fqutishat - You should mention this mechanism of setting the list index with @mavarley - presumably we need to know the interoperable mechanism for this.

Also - what is the behavior when this field is missing?

Reply:
verification will fail when revocationListIndex is missing. I add unit test.